stetho/src/main/java/com/facebook/stetho/inspector/network/AsyncPrettyPrinterSchemaFileManager.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.network;
+
+import android.content.Context;
+import com.facebook.stetho.common.LogRedirector;
+import com.facebook.stetho.common.Util;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class AsyncPrettyPrinterSchemaFileManager {
+  private static final String TAG = "AsyncPrettyPrinterSchemaCacheManager";
+  private static final String FILENAME_PREFIX = "async-pretty-printer-schema-";
+  private static int cacheFileCounter = 0;
+  private final Map<URL,String> mFilenameMap = Collections.synchronizedMap(
+      new HashMap<URL,String>());
+
+  private final Context mContext;
+
+  public AsyncPrettyPrinterSchemaFileManager(Context context) {
+    mContext = context;
+  }
+
+  public synchronized OutputStream openNewSchemaCache(URL schemaUrl)
```

Comment:
we can avoid making this method synchronized, as well as `readCache`, if we instead use an AtomicInteger for the cacheFileCounter